src/vanilla/server.rs

```diff
@@ -104,3 +140,6 @@ impl Server {
         Ok(())
     }
 }
+
+unsafe impl Send for Server {}
+unsafe impl Sync for Server {}
```

Comment:
Undocumented unsafe code is an absolutely unacceptable. Do not merge until this is fixed.